sdk/schemaregistry/schema-registry-avro/samples/v1-beta/javascript/schemaRegistryAvroSample.js

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Demonstrates the use of AvroSerializer to create messages with avro-serialized payload using schema from Schema Registry.
+ */
+
+const { DefaultAzureCredential } = require("@azure/identity");
+const { SchemaRegistryClient } = require("@azure/schema-registry");
+const { AvroSerializer } = require("@azure/schema-registry-avro");
+
+// Load the .env file if it exists
+require("dotenv").config();
+
+// The fully qualified namespace for schema registry
+const schemaRegistryFullyQualifiedNamespace =
+  process.env["SCHEMA_REGISTRY_ENDPOINT"] || "<endpoint>";
+
+// The schema group to use for schema registeration or lookup
+const groupName = process.env["SCHEMA_REGISTRY_GROUP"] || "AzureSdkSampleGroup";
+
+// Sample Avro Schema for user with first and last names
+const schemaObject = {
+  type: "record",
+  name: "User",
+  namespace: "com.azure.schemaregistry.samples",
+  fields: [
+    {
+      name: "firstName",
+      type: "string",
+    },
+    {
+      name: "lastName",
+      type: "string",
+    },
+  ],
+};
+
+const schema = JSON.stringify(schemaObject);
+
+// Description of the schema for registration
+const schemaDescription = {
+  name: `${schemaObject.namespace}.${schemaObject.name}`,
+  groupName,
+  format: "Avro",
+  definition: schema,
+};
+
+async function main() {
+  // Create a new client
+  const client = new SchemaRegistryClient(
+    schemaRegistryFullyQualifiedNamespace,
+    new DefaultAzureCredential()
+  );
+
+  // Register the schema. This would generally have been done somewhere else.
+  // You can also skip this step and let `serializeMessageData` automatically register
+  // schemas using autoRegisterSchemas=true, but that is NOT recommended in production.
+  await client.registerSchema(schemaDescription);
+
+  // Create a new serializer backed by the client
+  const serializer = new AvroSerializer(client, { groupName });
+
+  // serialize an object that matches the schema and put it in a message
+  const value = { firstName: "Jane", lastName: "Doe" };
+  const message = await serializer.serializeMessageData(value, schema);
```

Comment:
I must have missed this in the PR to switch to using Serializer, but we actually renamed this to just serialize in .NET. We also renamed MessageWithMetadata to BinaryContent. Is there a reason not to do this same change in JS?

Reply:
Ok I will talk to @bterlson about those renames.